msal/src/main/java/com/microsoft/identity/client/PublicClientApplicationConfiguration.java

```diff
@@ -78,6 +81,8 @@
 public class PublicClientApplicationConfiguration {
     private static final String TAG = PublicClientApplicationConfiguration.class.getSimpleName();
 
+    private static final Pattern BROKER_REDIRECT_URI_REGEX = Pattern.compile("msauth://([^/]*)/.*");
```

Comment:
Is this meant to be used outside of the context of an Android app? Because this regex seems to accept invalid android application package names

Reply:
Not really - it's private. But the idea is that this pulls out the authority segment of a URI with the correct scheme, and then we compare that string with equals on the package name. But there's only one usage of it...

Reply:
Okay, I think it is fine as this only being used in this class....but just wanted to point out that the group 1 which is supposed to match the package name, seems to accept anything but a slash, so I think this would accept redirect uri that are formed via package names that aren't actually valid Android application ids....but I think since this is all internal it is probably okay. However, it may still be good to make it more specific. 

FYI, Application id aka "Android app package name" naming conventions are listed here: https://developer.android.com/studio/build/application-id

Reply:
Yeah, I realize what it was doing... but my mind was in the regex prison, and we don't need to use a pattern at all.